docs/blueprints/EP038.rst

```diff
@@ -0,0 +1,71 @@
+
+Motivation
+==========
+
+The NApp **coloring** installs flows according to a switch neighbors (switches connected by a link). These flows have three crucial characteristics:
+
+  1. The matching criteria has one field, **"dl_src"** which has the same standard as a **MAC** address; ie. **ee:ee:ee:ee:ee:02**.
+  2. The received packet is send to the controller.
+  3. They have the highest priority.
+
+A customer connection with a certain **MAC** address could send packets indefinitely overloading the controller and interrupting other flows in the switch. Finally the currently used switch could disconnect.
+
+Solution
+========
+
+We need to ensure that the packets that **coloring** is interested in, come from inside the topology and drop anything else. Adding the following should suffice:
+
+  - Adding **"in_port"** as matching criteria to the created flow. The proper port could be obtained from the connection of switches neighbors.
+  - Add an extra flow without **"in_port"** so it matches what is left. The matched packets need to be dropped to avoid further inturreptions with other flows.
+
+Diagram on how the solution would look like:
+
+  .. code:: console
+
+                                                   +------------------------+ 
+                                                   |        Switch 1        | 
+                             Port 0         Port 0 |       (in_port=3       |  Port 3          Port 2  +-------------------------+ 
+    +--------------------+    /|     attack    |\  |dl_src=ee:ee:ee:ee:ee:02|   /|                |\   |         Switch 2        | 
+    |      Customer       |  / |---------------| \ |  actions=CONTROLLER)   |  / |----------------| \  |        (in_port=2       | 
+    |MAC=ee:ee:ee:ee:ee:02|  \ |---------------| / |                        |  \ |----------------| /  | dl_src=ee:ee:ee:ee:ee:01| 
+    +---------------------+   \|    packets->  |/  |           (            |   \|                |/   |   actions=CONTROLLER)   | 
+                                                   |dl_src=ee:ee:ee:ee:ee:02|                          +-------------------------+ 
+                                                   |      actions=drop)     | 
+                                                   +------------------------+ 
+
+Priority
+~~~~~~~~
+
+This solution is not going to stop loads of packets comming from the customer but just manage them. In order to not drop a necessary packet and free the rest of the flows from these attacks, the priorities need to be adjusted.
+
+- Leave the priority **50 000** for the **coloring** flow entries with **"in_port"** match in them
+- The flow entries that drop packet should have a higher priority than **mef_eline** entries (max. **20 000**) and lower than **of_lldp** entries (currently as of writting **1 000**)
```

Comment:
Priority for `of_lldp` flows is related to its issue [#85](https://github.com/kytos-ng/of_lldp/issues/85)